### PR TITLE
[CDS-100237] Added Google Cloud Run step details

### PR DESCRIPTION
**Pull request by vishal-av**
Thanks for contributing to the Harness Developer Hub! Our code owners will review your submission.

## Description

* Please describe your changes: \__________________________________
* Jira/GitHub Issue numbers (if any): \______________________________
* Preview links/images (Internal contributors only): \__________________

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [ ] Successful preview build.
- [ ] Code owner review.
- [ ] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

File: docs/continuous-delivery/deploy-srv-diff-platforms/google-cloud-functions/google-cloud-run.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+## Deployment Summary
+
+Here's a high-level summary of the setup steps:
+
+1. Create a Harness CD pipeline.
+2. Add a Deploy stage.
+3. Select the deployment type **Google Cloud Run**, and then select **Set Up Stage**.
+   ![](static/google-cloud-run-1.png)
+4. Select **Add Service**.
+   1. Add the **Google Cloud Run Manifest Definition** to the new Cloud Run service.
```
Comment (lovish1999) [inline docs/continuous-delivery/deploy-srv-diff-platforms/google-cloud-functions/google-cloud-run.md:28]: Google Cloud Run Service Manifest Definition

File: docs/continuous-delivery/deploy-srv-diff-platforms/google-cloud-functions/google-cloud-run.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+   3. **Google Cloud Run Deploy Step**:
+      - Applies the configurations YAML file and creates a new service or revision.
+      - The Deploy Step uses the `gcloud run services replace` command to apply the YAML manifest provided by the user. For more information, go to [Google Cloud Run Documentation](https://cloud.google.com/sdk/gcloud/reference/run/services/replace).
+      - Updates the container image field in the YAML manifest with the image provided in the service artifact, ensuring that the manifest reflects the correct configuration.
+      - You can provide additional commands in the **Optional Configuration** under **Google Cloud Run Services Replace Command Options**.
+      - Harness fetches the instance details configured by Google platform using the **Google Cloud Monitoring SDK** and saves them. These details can be viewed in the logs while running this step.
+      **Note**: We do not use the Google Run Deploy command here as this command takes every field as a command flag. Instead, we use the replace command and replace all the configurations provided in the YAML file.
```
Comment (lovish1999) [inline docs/continuous-delivery/deploy-srv-diff-platforms/google-cloud-functions/google-cloud-run.md:55]: Note should not be part of this point. It should be in the next line.

File: docs/continuous-delivery/deploy-srv-diff-platforms/google-cloud-functions/google-cloud-run.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+      Downloads the manifest specified in the service. 
+   2. **Google Cloud Run Prepare Rollback**:
+      Fetches the current configurations of the deployment and saves it. These details can be used during the rollback step.
+      - The Prepare Rollback step uses the `gcloud run services describe` command to fetch the details of the service. For more information, go to [Google Cloud Run Documentation](https://cloud.google.com/sdk/gcloud/reference/run/services/replace).
+   3. **Google Cloud Run Deploy Step**:
+      - Applies the configurations YAML file and creates a new service or revision.
+      - The Deploy Step uses the `gcloud run services replace` command to apply the YAML manifest provided by the user. For more information, go to [Google Cloud Run Documentation](https://cloud.google.com/sdk/gcloud/reference/run/services/replace).
+      - Updates the container image field in the YAML manifest with the image provided in the service artifact, ensuring that the manifest reflects the correct configuration.
+      - You can provide additional commands in the **Optional Configuration** under **Google Cloud Run Services Replace Command Options**.
```
Comment (lovish1999) [inline docs/continuous-delivery/deploy-srv-diff-platforms/google-cloud-functions/google-cloud-run.md:53]: command options

File: docs/continuous-delivery/deploy-srv-diff-platforms/google-cloud-functions/google-cloud-run.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+      - Harness fetches the instance details configured by Google platform using the **Google Cloud Monitoring SDK** and saves them. These details can be viewed in the logs while running this step.
+      **Note**: We do not use the Google Run Deploy command here as this command takes every field as a command flag. Instead, we use the replace command and replace all the configurations provided in the YAML file.
+   4. **Google Cloud Run Traffic Shift**:
```
Comment (lovish1999) [inline docs/continuous-delivery/deploy-srv-diff-platforms/google-cloud-functions/google-cloud-run.md:56]: Traffic Shift Step

File: docs/continuous-delivery/deploy-srv-diff-platforms/google-cloud-functions/google-cloud-run.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+9. Define the rollback step:
+   - **Rollback Step**:
+     - Reverts services to previous configurations or states.
+     - If it’s the first deployment, the service is deleted using the `gcloud run services delete` command.
+     - For subsequent deployments, traffic is redirected to older revisions using the `gcloud run services update-traffic` command.
+     **Note**: Google Cloud Run does not allow you to delete the new revision. Only the traffic is diverted to the previous revisions.
+
+10. Select **Save**, and then run the pipeline.
+
+## Cloud Functions Permission Requirements
```
Comment (lovish1999) [inline docs/continuous-delivery/deploy-srv-diff-platforms/google-cloud-functions/google-cloud-run.md:70]: Cloud Run

**Review by krishi0408 — APPROVED** [review]

**Pull request merged**